File: TrelloApp/src/app/core/models/comment.ts
import {Card} from './card';
import {User} from './user';

export interface Comment {
  id: number,
  text: string
  cardId: number,
  authorId: number,
  createdAt: string,
  updatedAt?: string,
  card: Card,
  user: User
}

export interface AddComment {
  text: string;
}

export interface UpdateComment {
  text: string;
}

File: TrelloApi/TrelloApi/src/Domain/Entities/Comment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TrelloApi.Domain.Entities;

[Table("Comment")]
public class Comment: Entity
{
    [StringLength(256), Required]
    public string Text { get; set; }
    
    [ForeignKey("Card"), Required]
    public int CardId { get; set; }
    public Card Card { get; set; }
    
    [ForeignKey("User"), Required]
    public int AuthorId { get; set; }
    public User User { get; set; }

    public Comment(string text, int cardId, int authorId)
    {
        Text = text;
        CardId = cardId;
        AuthorId = authorId;
    }
}